statedb/statedb.go

```diff
@@ -403,6 +404,29 @@ func (s *StateDB) TreeView(root []byte) (*TreeView, error) {
 	return &TreeView{
 		db:   s.db,
 		tree: tree,
-		cfg:  mainTreeCfg,
+		cfg:  MainTreeCfg,
 	}, nil
 }
+
+func (s *StateDB) Import(cfg TreeConfig, r io.Reader) (err error) {
+	dbtx := s.db.WriteTx()
+	defer func() {
+		if err != nil {
+			dbtx.Discard()
+		}
+	}()
+
+	var txTree db.WriteTx
+	if cfg.kindID == "" { // TreeMain
+		txTree = subWriteTx(dbtx, subKeyTree)
+	} else {
+		tx := subWriteTx(dbtx, path.Join(subKeySubTree, cfg.prefix))
+		txTree = subWriteTx(tx, subKeyTree)
+	}
+	tree, err := tree.New(txTree,
+		tree.Options{DB: nil, MaxLevels: cfg.maxLevels, HashFunc: cfg.hashFunc})
+
+	tree.ImportDumpReaderWithTx(txTree, r)
+	txTree.Commit()
```

Comment:
Review the error handling for this function please

Reply:
you're so pessimistic, how could any of this return an error? :rofl: 

thanks for the catch, fixed in 3628c22e